python/HeaderService/hutils.py

```diff
@@ -354,23 +338,31 @@ def calculate_string_header(self):
             hstring = hstring + str(self.header[extname]) + '\n' + self.hdu_delimiter
         return hstring
 
-    def write_headers(self, filenames, hstring, MP=False, NP=2, md5=False):
+    def write_header_yaml(self, filename):
+        """Write a header file in yaml format"""
 
-        """
-        Write one header per CCD (all the same) for now
-        in order to test I/O performance
-        """
-        if MP:
-            pool = multiprocessing.Pool(processes=NP)
-            args = [(filename, hstring, md5) for filename in filenames]
-            pool.map(write_header_string, args)
-            pool.close()
-            pool.join()
-        else:
-            for filename in filenames:
-                arg = filename, hstring, md5
-                write_header_string(arg)
-        return
+        # The dict where we will store the header contents
+        self.yaml_header = {}
```

Comment:
Does this need to be assigned to `self`? It would be less typing below to use a short local variable and then assign to `self` when the loop is finished.  This also means that if something goes wrong in the loop you don't end up with a partially populated `yaml_header`.

Reply:
No need really. I changed to simply `yaml_header`